Add SSM Installer Policy Pack (#795)
turbot/guardrails-samples: policy_packs/aws/ssm/enforce_install_packages_on_instances_using_ssm_package_installer/README.md

---
categories: ["security"]
primary_category: "security"
---

> # Enforce Install Packages On Instances Using AWS SSM Package Installer

Reviewer: ```suggestion
# Enforce Install Packages on AWS EC2 Instances Using AWS SSM Package Installer
```

I don't think `on` is capitalized in sentence case. Also, do we have any other titles where we have 2 or more services? Do we always include the cloud provider name, or just for the first service?

Reviewer: The dir name and file names should be updated too

Author: Updated.

Reviewer: @SumitPopat Do we need `AWS` twice? I'm still not sure on that part. Do we have any other examples of multiple services in a title?

> Enforcing the installation of packages on instances using the AWS SSM Package Installer is crucial for maintaining a consistent, secure, and automated software deployment process. This measure ensures that all package installations are managed and logged through AWS Systems Manager, enhancing security, compliance, and operational efficiency by providing centralized control and monitoring of software installations.

Reviewer: ```suggestion
Enforcing the installation of packages on EC2 instances using the AWS SSM Package Installer is crucial for maintaining a consistent, secure, and automated software deployment process. This measure ensures that all package installations are managed and logged through AWS Systems Manager, enhancing security, compliance, and operational efficiency by providing centralized control and monitoring of software installations.
```

We should be clear on what type of instances we're talking about, since multiple services have instance types, like Lightsail.

Author: Updated.

This [policy pack](https://turbot.com/guardrails/docs/concepts/resources/smart-folders) is designed to allow package installation on instances with or without direct access to original package source and can help you configure the following settings for SSM documents:

- Set the region ARN where the SSM resources will be deployed
- Set the IAM role ARN to be used by SSM to run documents
- Create three SSM documents in the specified region
  - Turbot-LinuxPackageInstaller-PublicNetwork
  - Turbot-LinuxPackageInstaller-PrivateNetwork
  - Turbot-LinuxS3PackageInstaller

## Documentation

- **[Review Policy settings →](https://hub-guardrails-turbot-com-git-development-turbot.vercel.app/policy-packs/enforce_install_packages_on_instances_using_ssm_package_installer/settings)**
- Using the SSM documents
  - There are two entry point types:
    - Turbot-LinuxPackageInstaller-PublicNetwork
    - Turbot-LinuxPackageInstaller-PrivateNetwork

> ### Turbot-LinuxPackageInstaller-PublicNetwork

Reviewer: Why are these under the documentation section? A follow-up question is what am I meant to do with this information?

Author: This was added for informational purpose on what the resources would be used for once created. Felt like it was useful information and copied directly from the existing readme for this stack example.

Use this entry point type when there is direct access to the package source.
Using this method, all that is required to the URL of the package to install.

### Turbot-LinuxPackageInstaller-PrivateNetwork

Use this entry point type when there is no direct access to the package source.
Using this method, it is required that the URL of the package to install and a valid S3 bucket which be used to cache the package.

### Turbot-LinuxS3PackageInstaller

This document should never be directly run but is a helper document to install the package.

### Current limitations

- Currently only Ubuntu is supported. Add support to more Linux flavours and Windows
- Handle package dependencies when installing through S3
- Test edge cases

## Getting Started

### Requirements

- [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli)
- Guardrails mods:
  - [@turbot/aws-ssm](https://hub-guardrails-turbot-com-git-development-turbot.vercel.app/aws/mods/aws-ssm)

> - AWS configuration to run the stack control:

Reviewer: Is it an AWS configuration or set of resources we need to run the stack control? Also, how are these resources used?

>   - S3 Bucket to store packages

Reviewer: ```suggestion
  - S3 bucket to store packages
```

I don't think we should capitalize resource names, as we don't below, e.g., SSM document, A role

Author: Updated.

>   - A role for SSM document execution with the following configured permissions:

Reviewer: ```suggestion
  - An IAM role for SSM document execution with the following configured permissions:
```

We should be consistent in using `service resource` in our text

Author: Updated

>     - SSM

Reviewer: Instead we should just give a sample policy that users can copy and paste?

Author: Added the same policy.

      - DescribeInstanceInformation
      - ListCommandInvocations
      - ListCommands
      - SendCommand
    - S3
      - PutObject

>   - A role for the instance with the following configured polices:

Reviewer: ```suggestion
  - An IAM role for the EC2 instance with the following configured polices:
```

Author: Updated.
